File: backend/app/src/main/java/com/yagubogu/widget/repository/WidgetDeviceRepository.java
package com.yagubogu.widget.repository;

import com.yagubogu.member.domain.Member;
import com.yagubogu.widget.domain.WidgetDevice;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface WidgetDeviceRepository extends JpaRepository<WidgetDevice, Long> {

    Optional<WidgetDevice> findByDeviceId(String deviceId);

    Optional<WidgetDevice> findByDeviceIdAndMemberId(String deviceId, Long memberId);

    void deleteAllByMember(Member member);
}

File: mobile/composeApp/src/commonMain/kotlin/com/yagubogu/data/repository/widget/WidgetDeviceRepository.kt
package com.yagubogu.data.repository.widget

interface WidgetDeviceRepository {
    suspend fun registerDevice(pushToken: String): Result<Unit>

    /** 현재 기기의 디바이스 등록을 해제한다. */
    suspend fun deregisterDevice(): Result<Unit>
}
